source/geometries/NextDemoSiPMBoard.cc

```diff
@@ -79,6 +85,40 @@ NextDemoSiPMBoard::~NextDemoSiPMBoard()
 void NextDemoSiPMBoard::Construct()
 {
 
+  G4double sipm_z_dim = 0.;
+
+  if (sipm_type_ == "sensl"){
+    SiPMSensl* sipm = new SiPMSensl();
+
+    sipm->SetVisibility(sipm_visibility_);
+    sipm->SetTimeBinning(time_binning_);
+    sipm->SetSensorDepth(3);
+    sipm->SetMotherDepth(5);
+    sipm->SetNamingOrder(1000);
+    sipm->Construct();
+
+    sipm_ = sipm;
+
+    sipm_z_dim = sipm->GetDimensions().z();
+  }
+  else if (sipm_type_ == "next100"){
+    Next100SiPM* sipm = new Next100SiPM();
+
+    sipm->SetVisibility(sipm_visibility_);
+    sipm->SetTimeBinning(time_binning_);
+    sipm->SetSensorDepth(2);
+    sipm->SetMotherDepth(4);
+    sipm->SetNamingOrder(1000);
+    sipm->Construct();
+
+    sipm_ = sipm;
+
+    sipm_z_dim = sipm->GetDimensions().z();
+  }
+  else
+    G4Exception("[NextDemoSiPMBoard]", "Construct()",
```

Comment:
Same comment as before: I'm not sure we need this exception, since the value of `sipm_type_` is set automatically when the configuration parameter for the run is chosen.

Reply:
I added this exception together with the hole equivalent in
https://github.com/gondiaz/nexus/blob/8e8a377656b9fb50d0fc14c0116cefeeb9b8a5a8/source/geometries/NextDemoTrackingPlane.cc#L158

Maybe is useful if someone is about to change the parameters for some run (which is not a good idea by the way) or to add another one with different parameters. Both checks are only made once.

Reply:
As I see it, we want to have the configurations frozen, with `runN` having all its correct parameters. Checking that the sipm type is one of those doesn't really ensure that it is the correct type, just that is one of the possible types. Therefore, I'm not sure this check is useful, although is not harmful, either.

Reply:
Agree, to me sounds better to keep those parameters constant and correct for each specific run configuration. I remove the conditionals.